fdbclient/ClusterInterface.h

```diff
@@ -119,13 +119,14 @@ struct OpenDatabaseRequest {
 	Arena arena;
 	StringRef issues, traceLogGroup;
 	VectorRef<ClientVersionRef> supportedVersions;
+	bool client_tls_configured;
 	UID knownClientInfoID;
 	ReplyPromise< struct ClientDBInfo > reply;
 
 	template <class Ar>
 	void serialize(Ar& ar) {
 		ASSERT( ar.protocolVersion() >= 0x0FDB00A400040001LL );
-		serializer(ar, issues, supportedVersions, traceLogGroup, knownClientInfoID, reply, arena);
+		serializer(ar, issues, supportedVersions, client_tls_configured, traceLogGroup, knownClientInfoID, reply, arena);
```

Comment:
I don't know the details about the deserialization part but in general I think it's a better idea to `append` this instead of `insert` so that gives the receiver who's running one an older code base a chance to simple ignore the bytes after what it can understand.

Reply:
Hmm, I think it is good to let the system crash immediately if the protocol is inconsistent than letting it run. This help avoid the weird bugs.

Reply:
That will be the protocol when the FlatBuffers/Object serializer work is done, and it'll be required to only append items.  However, until that happens, historically we've keep the serialization order the same as the member order, and any changes to it are gated on the protocol version to determine if a particular field in a position should be present or not.

Reply:
About " it'll be required to only append items", is it a recommendation by FlatBuffers or a hard requirement?